internal/bodyprocessors/json.go

```diff
@@ -18,26 +18,52 @@ type jsonBodyProcessor struct{}
 var _ plugintypes.BodyProcessor = &jsonBodyProcessor{}
 
 func (js *jsonBodyProcessor) ProcessRequest(reader io.Reader, v plugintypes.TransactionVariables, _ plugintypes.BodyProcessorOptions) error {
+	// Read the entire body to store it and process it
+	s := strings.Builder{}
+	tr := io.TeeReader(reader, &s)
+
+	// Process as normal
 	col := v.ArgsPost()
-	data, err := readJSON(reader)
+	data, err := readJSON(tr)
 	if err != nil {
 		return err
 	}
 	for key, value := range data {
 		col.SetIndex(key, 0, value)
 	}
+
+	// Store the raw JSON in the TX variable for validateSchema
+	// This is needed because RequestBody is a Single interface without a Set method
+	if txVar := v.TX(); txVar != nil && v.RequestBody() != nil {
+		// Store the content type and raw body
+		txVar.Set("json_request_body", []string{s.String()})
```

Comment:
shall we do this if and only if we are validating schema?

Reply:
I'm definitely open to suggestions- at the time the bodyprocessor runs I don't think we know yet whether or not we will be doing a schema validate? I could delay processing, but then the JSON isn't available for the existing functionality.

I definitely could be missing something here though - if you have an idea of how to do this otherwise I'd be happy to make adjustments.

Reply:
ping @jptosso @M4tteoP

Reply:
ping @jptosso @M4tteoP